experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -15,51 +15,206 @@
+ * control over metrics.
+ */
 export abstract class MetricReader {
+  // Tracks the shutdown state.
   private _shutdown = false;
+  // MetricProducer used by this instance.
   private _metricProducer?: MetricProducer;
 
-  constructor(private _exporter: MetricExporter) {}
+  constructor(private readonly _preferredAggregationTemporality = AggregationTemporality.CUMULATIVE) {
+  }
 
+  /**
+   * Set the {@link MetricProducer} used by this instance.
+   *
+   * @param metricProducer
+   */
   setMetricProducer(metricProducer: MetricProducer) {
     this._metricProducer = metricProducer;
+    this.onInitialized();
   }
 
+  /**
+   * Get the {@link AggregationTemporality} preferred by this {@link MetricReader}
+   */
   getPreferredAggregationTemporality(): AggregationTemporality {
-    return this._exporter.getPreferredAggregationTemporality();
+    return this._preferredAggregationTemporality;
   }
 
-  async collect(): Promise<void> {
+  /**
+   * Handle once the SDK has initialized this {@link MetricReader}
+   *
+   * @protected
+   */
+  protected abstract onInitialized(): void;
+
+  /**
+   * Handle a shutdown signal by the SDK.
+   *
+   * <p> For push exporters, this should shut down any intervals and close any open connections.
+   * @protected
+   */
+  protected abstract onShutdown(): Promise<void>;
+
+  /**
+   * Handle a force flush signal by the SDK.
+   *
+   * <p> In all scenarios metrics should be collected via {@link collect()}.
+   * <p> For push exporters, this should collect and report metrics.
+   * @protected
+   */
+  protected abstract onForceFlush(): Promise<void>;
+
+  /**
+   * Collect all metrics from the associated {@link MetricProducer}
+   */
+  collect(options: ReaderCollectionOptions): void {
+    const timeout = options.timeoutMillis ?? 10000;
+    const done = options.done ?? (_result => {
+    });
+
     if (this._metricProducer === undefined) {
-      throw new Error('MetricReader is not bound to a MeterProvider');
+      done({
+        code: ReaderResultCode.FAILED,
+        error: new Error('MetricReader is not bound to a MetricProducer'),
+      });
+      return;
     }
-    const metrics = await this._metricProducer.collect();
 
-    // errors thrown to caller
-    await this._exporter.export(metrics);
+    // Subsequent invocations to collect are not allowed. SDKs SHOULD return some failure for these calls.
+    if (this._shutdown) {
+      done({
+        code: ReaderResultCode.FAILED,
+        error: new Error('Collection is not allowed after shutdown'),
+      });
+      return;
+    }
+
+    promiseWithTimeout(this._metricProducer.collect(), timeout, done);
   }
 
-  async shutdown(): Promise<void> {
+  /**
+   * Shuts down the metric reader, the callback will fire after the specified timeout or after completion.
+   *
+   * <p> NOTE: this operation will continue even after the timeout fires the callback.
+   * @param options
+   */
+  shutdown(options: ReaderForceFlushOptions): void {
```

Comment:
Ditto, even BasicTracerProvider defines `forceFlush` and `shutdown` return value as `Promise<void>`.